worlds/pokemon_emerald/__init__.py

```diff
@@ -0,0 +1,832 @@
+                should_match_type = self.multiworld.static_encounters[self.player] in {
+                    RandomizeStaticEncounters.option_match_type,
+                    RandomizeStaticEncounters.option_match_base_stats_and_type
+                }
+
+                for encounter in emerald_data.static_encounters:
+                    original_species = self.modified_data.species[encounter.species_id]
+                    target_bst = sum(original_species.base_stats) if should_match_bst else None
+                    target_type = self.random.choice(original_species.types) if should_match_type else None
+
+                    self.modified_data.static_encounters.append(StaticEncounterData(
+                        get_random_species(self.random, self.modified_data.species, target_bst, target_type).species_id,
+                        encounter.rom_address
+                    ))
+
+        def randomize_opponent_parties() -> None:
+            should_match_bst = self.multiworld.trainer_parties[self.player] in {
+                RandomizeTrainerParties.option_match_base_stats,
+                RandomizeTrainerParties.option_match_base_stats_and_type
+            }
+            should_match_type = self.multiworld.trainer_parties[self.player] in {
+                RandomizeTrainerParties.option_match_type,
+                RandomizeTrainerParties.option_match_base_stats_and_type
+            }
+            allow_legendaries = self.multiworld.allow_trainer_legendaries[self.player].value == Toggle.option_true
+
+            for trainer in self.modified_data.trainers:
+                new_party = []
+                for pokemon in trainer.party.pokemon:
+                    original_species = emerald_data.species[pokemon.species_id]
+                    target_bst = sum(original_species.base_stats) if should_match_bst else None
+                    target_type = self.random.choice(original_species.types) if should_match_type else None
+
+                    new_species = get_random_species(
+                        self.random,
+                        self.modified_data.species,
+                        target_bst,
+                        target_type,
+                        allow_legendaries
+                    )
+
+                    # Could cache this per species
+                    tm_hm_movepool = list({
+                        self.modified_data.tmhm_moves[i]
+                        for i, is_compatible in enumerate(int_to_bool_array(new_species.tm_hm_compatibility))
+                        if is_compatible
+                    })
+                    level_up_movepool = list({
+                        move.move_id
+                        for move in new_species.learnset
+                        if move.level <= pokemon.level
+                    })
+
+                    new_moves = (
+                        self.random.choice(tm_hm_movepool if self.random.random() < 0.25 and len(tm_hm_movepool) > 0 else level_up_movepool),
```

Comment:
super trivial thing that's just kind of bothering me lmao.
```suggestion
                        self.random.choice(tm_hm_movepool if self.random.random() < 0.25 and tm_hm_movepool else level_up_movepool),
```
in python an empty list resolves to `False`

Reply:
I'm not a big fan of truthy and falsy values aside from the occasional int. Makes things less readable and slightly more prone to unintended behavior imo. Especially when you have to keep multiple languages straight; empty arrays are truthy in JS, for example.